### PR TITLE
Add hamming distance

### PR DESCRIPTION
**Pull request by saybarsnazlica**
### Describe your change:

I added the algorithm to calculate the hamming distance between two strings.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/hamming_distance.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    >>> hamming_distance("karolin", "kathrin")
+    3
+    >>> hamming_distance("00000", "11111")
+    5
+    """
+    assert len(string1) == len(string2)
+
+    count = 0
+
+    for i in range(len(string1)):
```
Comment (lachlantownshend) [inline strings/hamming_distance.py:25]: The more pythonic way of doing this is

```
for char1, char2 in zip(string1, string2):
    if char1 != char2:
        count += 1
```

Comment (saybarsnazlica) [reply strings/hamming_distance.py:25]: I see, thanks. I'll update it.

**Comment by saybarsnazlica** [comment]
@lachlantownshend Could you please review this pull request?

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

File: strings/hamming_distance.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        int: Hamming distance
+
+    >>> hamming_distance("python", "python")
+    0
+    >>> hamming_distance("karolin", "kathrin")
+    3
+    >>> hamming_distance("00000", "11111")
+    5
+    """
+    assert len(string1) == len(string2)
```
Comment (poyea) [inline strings/hamming_distance.py:21]: How about raising an exception if they're not equal, and add a test case to it?

Comment (saybarsnazlica) [reply strings/hamming_distance.py]: Yes, I added an exception. Thank you!

**Pull request merged**